src/main/java/org/spongepowered/api/text/title/TitleBuilder.java

```diff
@@ -24,91 +24,311 @@
+     * <p>The default value for Vanilla is 20 (1 second).</p>
      *
-     * @param ticks The amount of ticks (1/20 second) for the fade in effect
+     * @param fadeIn The amount of ticks (1/20 second) for the fade in effect,
+     *        or {@code null} to reset
      * @return This title builder
+     * @see Title#getFadeIn()
+     */
+    public TitleBuilder fadeIn(@Nullable Integer fadeIn) {
+        this.fadeIn = fadeIn;
+        return this;
+    }
+
+    /**
+     * Returns the current stay effect time of the title.
+     *
+     * @return The current stay time, or {@link Optional#absent()} if none
+     * @see Title#getStay()
      */
-    TitleBuilder fadeIn(int ticks);
+    public final Optional<Integer> getStay() {
+        return Optional.fromNullable(this.stay);
+    }
 
     /**
-     * Set the duration in ticks how long the title should stay on the screen.
+     * Sets the duration in ticks how long the title should stay on the screen.
      * Once this period of time is over the title will fade out using the
-     * duration specified in {@link #fadeOut(int)}.
+     * duration specified in {@link #fadeOut(Integer)}.
      *
-     * <p>
-     * The default value for Vanilla is 60 (3 seconds).
-     * </p>
+     * <p>The default value for Vanilla is 60 (3 seconds).</p>
      *
-     * @param ticks The amount of ticks (1/20 second) to stay
+     * @param stay The amount of ticks (1/20 second) to stay, or {@code null} to
+     *        reset
      * @return This title builder
+     * @see Title#getStay()
      */
-    TitleBuilder stay(int ticks);
+    public TitleBuilder stay(@Nullable Integer stay) {
+        this.stay = stay;
+        return this;
+    }
 
     /**
-     * Set the duration in ticks of the fade out effect of the title.
+     * Returns the current fade out effect time of the title.
      *
-     * <p>
-     * The default value for Vanilla is 20 (1 second).
-     * </p>
+     * @return The current fade out time, or {@link Optional#absent()} if none
+     * @see Title#getFadeOut()
+     */
+    public final Optional<Integer> getFadeOut() {
+        return Optional.fromNullable(this.fadeOut);
+    }
+
+    /**
+     * Sets the duration in ticks of the fade out effect of the title.
      *
-     * @param ticks The amount of ticks (1/20 second) for the fade out effect
+     * <p>The default value for Vanilla is 20 (1 second).</p>
+     *
+     * @param fadeOut The amount of ticks (1/20 second) for the fade out effect,
+     *        or {@code null} to reset
      * @return This title builder
+     * @see Title#getFadeOut()
+     */
+    public TitleBuilder fadeOut(@Nullable Integer fadeOut) {
+        this.fadeOut = fadeOut;
+        return this;
+    }
+
+    /**
+     * Returns whether this builder is currently configured to clear.
+     *
+     * @return {@code true} if the title will clear
+     * @see Title#isClear()
      */
-    TitleBuilder fadeOut(int ticks);
+    public final boolean isClear() {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (67/60 (111%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:55258 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.84)
	(envelope-from <noreply@github.com>)
	id 1YNeaS-0003ZD-Ib
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:37:00 +0000
Date: Tue, 17 Feb 2015 01:37:00 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1424165820;
	bh=S918lOA3V0sc/U6FiUgXvYCDm3GZ06zz2fMdErt8uBM=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=rAuoow0UCOAtizdXMjv9WxpJuwO5OLgiE9tvzVs6+sPfj8dB2Y1r+GINa0Bh0Akm7
	 eKYJP2wT/OTjI5XYknna/4RK5vpaxVPFC54z2MMrvuEAkJkMv3k4UH2vQRlnT16Ztr
	 7/ujFGflhKcVo8F9KYMMulgnm12MKZEjF/Ui79Nw=
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803116@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30bbc769e2_7b4e3fc728f0529c328020";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bf310a4563f5d8bd0d681d353e977f2155cd530d1192cf0000000110facdbc92a170ce017a772c@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAv1tPn7kbj_y3biAv1w-yki_8VnjFks5nswM8gaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk


----==_mimepart_54e30bbc769e2_7b4e3fc728f0529c328020
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

>       */
> -    TitleBuilder fadeOut(int ticks);
> +    public final boolean isClear() {

why `public final` here, of all places?!?

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803116
----==_mimepart_54e30bbc769e2_7b4e3fc728f0529c328020
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803116">src/main/java/org/spongepowered/api/text/title/TitleBuilder.java</a>:</p>
<pre style='color:#555'>&gt;       */
&gt; -    TitleBuilder fadeOut(int ticks);
&gt; +    public final boolean isClear() {
</pre>
<p>why <code>public final</code> here, of all places?!?</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803116">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv_JYfXkhsiIJqGcqFcmTjfuIDP9Qks5nswM8gaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803116"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30bbc769e2_7b4e3fc728f0529c328020--